### PR TITLE
Migrate from Travis to Azure

### PR DESCRIPTION
**Pull request by StrikerRUS**
Refer to https://github.com/BayesWitnesses/m2cgen/pull/114#discussion_r339281797.

Results can be observed here: https://dev.azure.com/nekit94-120479/nekit94-12/_build/results?buildId=6.

List of installed software on `ubuntu-16.04`: https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/linux/Ubuntu1604-README.md.

The only thing in needed to do is register Azure account (if not yet), create a **public** project and link it to this repo.

ping @krinart @izeigerman

### CODE REVIEW

File: requirements-test.txt [hunk 1]
```diff
@@ -9,3 +9,4 @@ pytest-mock==1.10.0
 coveralls==1.5.1
 pytest-cov==2.6.1
 py-mini-racer==0.1.18
+attrs==19.1.0
```
Comment (StrikerRUS) [inline requirements-test.txt]: Fix 
```
[command]/bin/bash --noprofile --norc /home/vsts/work/_temp/8eb70c1f-fd00-43a7-bea0-aa872dac2bfe.sh
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.5.7/x64/bin/pytest", line 10, in <module>
    sys.exit(main())
  File "/opt/hostedtoolcache/Python/3.5.7/x64/lib/python3.5/site-packages/_pytest/config/__init__.py", line 61, in main
    config = _prepareconfig(args, plugins)
  File "/opt/hostedtoolcache/Python/3.5.7/x64/lib/python3.5/site-packages/_pytest/config/__init__.py", line 182, in _prepareconfig
    config = get_config()
  File "/opt/hostedtoolcache/Python/3.5.7/x64/lib/python3.5/site-packages/_pytest/config/__init__.py", line 156, in get_config
    pluginmanager.import_plugin(spec)
  File "/opt/hostedtoolcache/Python/3.5.7/x64/lib/python3.5/site-packages/_pytest/config/__init__.py", line 530, in import_plugin
    __import__(importspec)
  File "/opt/hostedtoolcache/Python/3.5.7/x64/lib/python3.5/site-packages/_pytest/tmpdir.py", line 25, in <module>
    class TempPathFactory(object):
  File "/opt/hostedtoolcache/Python/3.5.7/x64/lib/python3.5/site-packages/_pytest/tmpdir.py", line 35, in TempPathFactory
    lambda p: Path(os.path.abspath(six.text_type(p)))
TypeError: attrib() got an unexpected keyword argument 'convert'
```
error.

Refer to https://github.com/pytest-dev/pytest/issues/3280#issuecomment-538031871.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  # Since we develop a library, we want to make sure that it properly runs
+  # after installation. We already ran into an issue when some files were
+  # forgotten to be included in package_data and it was not caught by tests.
+
+  # So first we run unit tests against raw source code and report coverage.
+
+  - script: pytest -v tests/ --cov=m2cgen/ --ignore=tests/e2e/
+    displayName: Pytest
+#  - script: coveralls
+#    displayName: Coveralls
```
Comment (StrikerRUS) [inline azure-pipelines.yml]: Should be configured manually (with token):
```
[command]/bin/bash --noprofile --norc /home/vsts/work/_temp/3e5c0a43-effc-4bd1-9192-d036a83503e2.sh
Not on Travis or CircleCI. You have to provide either repo_token in .coveralls.yml or set the COVERALLS_REPO_TOKEN env var.
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.5.7/x64/lib/python3.5/site-packages/coveralls/cli.py", line 60, in main
    service_name=options['--service'])
  File "/opt/hostedtoolcache/Python/3.5.7/x64/lib/python3.5/site-packages/coveralls/api.py", line 59, in __init__
    self.ensure_token()
  File "/opt/hostedtoolcache/Python/3.5.7/x64/lib/python3.5/site-packages/coveralls/api.py", line 68, in ensure_token
    self.config_filename))
coveralls.exception.CoverallsException: Not on Travis or CircleCI. You have to provide either repo_token in .coveralls.yml or set the COVERALLS_REPO_TOKEN env var.
```

**Comment by krinart** [comment]
I think migrating to Azure should be considered as a last resort when there's literally no way to fix the original problem. To me it doesn't seem to be the case, at least for now.

As I mentioned in this comment https://github.com/BayesWitnesses/m2cgen/pull/114#discussion_r339282850, sometimes we need an ability to run e2e tests locally, and I don't think we should have e2e tests running for more than 50 minutes.

Before making this change I would kindly ask you to investigate what exactly takes so much time. You could start by looking into single model and comparing it to other languages.

And thanks for your constant initiative to make m2cgen better, we really appreciate your effort!

**Comment by StrikerRUS** [comment]
@krinart 

>  Before making this change I would kindly ask you to investigate what exactly takes so much time. You could start by looking into single model and comparing it to other languages.

Unfortunately, I can only confirm that PowerShell is quite slow. Tests for all models run significant amount of time, and there is no "one test that takes the most time of 50 minutes". I guess it's specific to PowerShell and nothing can be done with it.

**Comment by StrikerRUS** [comment]
OK, I think I'd better close this PR for now. Anyway, you can take my proposed config from the `Files changed` tab any moment you need it in the future.

**Pull request closed without merging**